core/script/mixin_script/editor/mixin_script_editor.cpp

```diff
@@ -155,10 +156,16 @@ void MixinScriptEditor::_on_editor_script_changed(Ref<Script> p_script) {
 	}
 	// Otherwise, delegate editing to other script editors automatically.
 	Ref<MixinScript> ms = p_script;
-	if (ms.is_valid() && ms->get_mixin_count() > 0) {
-		Ref<Script> first_script = ms->get_mixin(0);
-		if (first_script.is_valid()) {
-			ScriptEditor::get_singleton()->edit(first_script);
+	if (EditorSettings::get_singleton()->get_setting("text_editor/files/open_first_script_on_editing_mixin_script")) {
```

Comment:
Godot provides standard way to get and initialize settings:

```suggestion
	if (bool(EDITOR_GET("text_editor/files/open_first_script_on_editing_mixin_script"))) {
```

Reply:
Fixed in the last push